aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -203,18 +203,18 @@ struct AbsMaxOps {
   }
 
 #if defined(__CUDACC__) || defined(__HIPCC__)
-  inline C10_DEVICE acc_t warp_shfl_down(acc_t data, int offset) const {
-    return WARP_SHFL_DOWN(data, offset);
+  inline C10_DEVICE acc_t warp_shfl_down(acc_t acc, int offset) const {
+    return WARP_SHFL_DOWN(acc, offset);
   }
 #endif
 };
 
-template <typename acc_t>
+template <typename scalar_t, typename acc_t=scalar_t>
 struct NormOps {
   acc_t norm_;
 
-  inline C10_DEVICE acc_t reduce(acc_t acc, acc_t data, int64_t /*idx*/) const {
-    return acc + compat_pow(std::abs(data), norm_);
+  inline C10_DEVICE acc_t reduce(acc_t acc, scalar_t data, int64_t /*idx*/) const {
+    return acc + compat_pow(acc_t(std::abs(data)), norm_);
```

Comment:
What does the `acc_t` do here? Is it equivalent to `static_cast<acc_t>(...)`?

Reply:
Yeah this is a C-style cast. But it's probably better to use `static_cast` (https://anteru.net/blog/2007/c-background-static-reinterpret-and-c-style-casts/), so I'll go ahead and change all of these.